lightningd/plugin.h

```diff
@@ -1,28 +1,88 @@
 #ifndef LIGHTNING_LIGHTNINGD_PLUGIN_H
 #define LIGHTNING_LIGHTNINGD_PLUGIN_H
 #include "config.h"
+#include <ccan/intmap/intmap.h>
+#include <ccan/io/io.h>
 #include <ccan/take/take.h>
 #include <ccan/tal/tal.h>
 #include <lightningd/jsonrpc.h>
 #include <lightningd/log.h>
 
+/**
+ * A plugin, exposed as a stub so we can pass it as an argument.
+ */
+struct plugin {
```

Comment:
Why is it necessary to make all these things public? Wouldn't it be sufficient to make `plugin_kill` public?

Reply:
I think I need it for https://github.com/darosior/lightning/blob/99b4686406d5c79761cd03fe111b1140016afbe7/lightningd/plugin_control.c#L26